### DISCUSSION TITLE
Wrong JAVA version

### DISCUSSION THREAD

**Discussion by F42J** [itzg/docker-minecraft-server#2676] [Q&A] [answered]
I tried starting the server using a custom modpack supplied as curseforge manifest and additionally downloaded missing mods that couldnt be auto loaded

However the server fails on startup with the following error
The set properties are:
TYPE=AUTO_CURSEFORGE
CF_SLUG=custom
CF_MODPACK_MANIFEST=<path/to/tile>
CF_API_KEY=<<key>

Error:
Exception in thread "main" java.lang.IllegalAccessError: class cpw.mods.modlauncher.SecureJarHandler (in unnamed module @0x404b9385) cannot access class sun.security.util.ManifestEntryVerifier (in module java.base) because module java.base does not export sun.security.util to unnamed module @0x404b9385

From some initial troubleshooting, it looks like the problem is that forge depends oń some java features removed in java 16. So forge will only run in versions up to java 15, but it seems the container uses a newer one.

Is there any way to force the container to use an older java version e.g java 15 to run forge correctly?

Thanks in advance for any help on this

**Comment by itzg** [comment]
https://docker-minecraft-server.readthedocs.io/en/latest/versions/java/

**Reply by F42J** [reply]
Thanks for the quick reply, this solved the problem.
Unfortunately the setup still fails when installing the modpack as a client side mod is loaded.
I tried excluding the mod as said in the doumentation using the following line with its id. however, the mod doesnt seem to be excluded

CF_EXCLUDE_MODS: "441024"

Thanks in advance for the help

**Reply by itzg** [reply, marked as answer]
Make sure you forced synchronize to pick up that exclusion:

https://docker-minecraft-server.readthedocs.io/en/latest/types-and-platforms/mod-platforms/auto-curseforge/#unpublished-modpacks:~:text=If%20needing%20to%20iterate%20on%20the%20options%20above%2C%20set%20CF_FORCE_SYNCHRONIZE%20to%20%22true%22%20to%20ensure%20the%20exclude/includes%20are%20re%2Devaluated.
